components/kyma-environment-broker/internal/process/provisioning/event_hub_step.go

```diff
@@ -140,6 +139,77 @@ func (p *ProvisionAzureEventHubStep) Run(operation internal.ProvisioningOperatio
 	return operation, 0, nil
 }
 
+func (p *ProvisionAzureEventHubStep) resourcesExist(resourcesName string, tags azure.Tags, azureClient azure.AzureInterface) (bool, error) {
+	exists, err := azureClient.ResourceGroupExists(p.context, resourcesName, tags)
+	if err != nil {
+		return false, errors.Wrapf(err, "Can't check Azure resource group %s.", resourcesName)
+	}
+	// we are using the same name for resource group and namespace name
+	if exists {
+		exists, err = azureClient.NamespaceExists(p.context, resourcesName, resourcesName, tags)
+		if err != nil {
+			return false, errors.Wrapf(err, "Can't check Azure Event Hubs namespace %s.", resourcesName)
+		}
+	}
+	return exists, nil
```

Comment:
i really like early returns
```suggestion
	exists, err := azureClient.ResourceGroupExists(p.context, resourcesName, tags)
	if err != nil {
		return false, errors.Wrapf(err, "Can't check Azure resource group %s.", resourcesName)
	}
	// we are using the same name for resource group and namespace name
	if ! exists {
	         return false, nil
	}
	exists, err = azureClient.NamespaceExists(p.context, resourcesName, resourcesName, tags)
	if err != nil {
		return false, errors.Wrapf(err, "Can't check Azure Event Hubs namespace %s.", resourcesName)
	}
	return exists, nil
```